src/Aspire.Hosting.Azure/AzureOpenAIResource.cs

```diff
@@ -11,20 +11,25 @@ namespace Aspire.Hosting.ApplicationModel;
 /// <param name="name">The name of the resource.</param>
 public class AzureOpenAIResource(string name) : Resource(name), IAzureResource, IResourceWithConnectionString
 {
-    private readonly Collection<AzureOpenDeploymentResource> _deployments = [];
+    private readonly Collection<AzureOpenAIDeploymentResource> _deployments = [];
```

Comment:
`Collection<T>`? Why not `List<T>`

Reply:
@davidfowl I followed the way `AzureSqlServerResource` was implemented since it had a similar model. Should I go ahead and replace this by IList/IReadOnlyList in both resources?